### ISSUE TITLE
How quickly do HTMX elements bind to document events?

### ISSUE THREAD

**Issue by deanebarker** [bigskysoftware/htmx#1799]
Is there any delay in binding elements to events?

I have a textbox defined like this (note the binding to the `doSearch` event at the end of the second line):

```
<input type="text" id="q" name="q"
 hx-get="/search" hx-trigger="keyup changed delay:300ms, doSearch from:body"
 hx-target="#results-container" />
```

Inside the `DOMContentLoaded` event, I have this:

```
document.body.dispatchEvent(new Event('doSearch'))
```

It does not trigger on page load. I can debug through it, and watch the call to `dispatchEvent` which returns `true`, but the request doesn't go out. After the page has loaded, if I put the above JS in the console and run it, it works fine.

So:

1. I know the event calling code is correct -- it works from the console
2. I know the event is being raised -- it returns `true` (I printed it to the console)
3. I know the HTMX is correct -- again, it works when the event is raised from the console

My only thought is that when the event is raised inside `DOMContentLoaded`, the input element hasn't been bound to the event yet? So, the event is happening, but the input doesn't yet realize it's supposed to respond? Any validity to that theory?

**Comment by andryyy** [comment]
I think you’d need to enclose it in a htmx.onLoad(). 

Have you tried adding "load, " in front of your other triggers?

**Comment by deanebarker** [comment]
Nope. I tried that. `html:load` fires ever y time HTMX adds content to the DOM, not when the library loads.

**Comment by andryyy** [comment]
"load once" should work then.

**Comment by andryyy** [comment]
And if a htmx event reinjects the element, a htmx:afterSettle could be useful, too. But on a regular load, it will only fire "on load".

**Comment by 1cg** [comment]
no longer relevant
